File: CodeChef/aug15/grguy.py
l = []
def tryrun(li):
	i = ct = 0
	while l[li][i]:
		if l[li][i] == '#':
			return 9223372036854775807
		while l[li][i] == '.':
			i+=1
		li = not(li)
		ct+=1
	return ct-1

t = int(raw_input())
while t>0:
	t-=1
	l1 = raw_input()
	l[0] = list(l1)
	l2= raw_input()
	l[1] = list(l2)
	a = min(tryrun(0), tryrun(1))
	if a == 9223372036854775807:
		print "No"
	else:
		print "Yes"
		print a

File: CodeChef/aug15/grguy.cpp
#include <bits/stdc++.h>
using namespace std;

#define max INT_MAX;

int main() 
{
	ios_base::sync_with_stdio(false);
	int t;
	cin>>t;

	while(t--)
	{
		string l[2];
		cin>>l[0];
		cin>>l[1];
		
		int temp1 = 0, z = 0, i = 0, kz = 0;

		while(l[0][i]) 
		{
			if(l[z][i] == '#') 
			{
				temp1 = max;
				goto x;
			}
			while(l[z][i] == '.') 
				i++;
			z = !z; 
			kz++;
		}
		temp1 = kz - 1;
		x:

		int temp2 = 0;
		z = 1;
		i = 0;
		kz = 0;
		while(l[0][i]) 
		{
			if(l[z][i] == '#') 
			{
				temp2 = max;
				goto l;
			}
			while(l[z][i] == '.') 
				i++;
			z = !z; 
			kz++;
		}
		temp2 = kz - 1;
		l:

		if(min(temp1,temp2) == INT_MAX) 
			cout<<"No"<<endl;
		else 
			cout<<"Yes"<<endl<<min(temp1,temp2)<<endl;
	}
	return 0;
}